File: leetcode_C++/p1572.cpp
/*
@Author: mrxhar
@Date:   2023-09-13 10:44:23
@Email:  1337904145@qq.com
@Blog:   https://leetcode.cn/problems/matrix-diagonal-sum/?envType=daily-question&envId=2023-09-13
*/
#include <vector>
using namespace std;

class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int n = mat.size();
        int res = 0;
        for (int i = 0; i < n; i++) {
            res += (mat[i][i] + mat[i][n - i - 1]);
        }
        return res - (n % 2) ? mat[n / 2][n / 2] : 0;
    }
};

File: leetcode_py/p1572.py
from typing import List


class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        res = 0
        n = len(mat)
        for i in range(n):
            res += (mat[i][i] + mat[i][n - 1 - i])
        return res - (n % 2 and mat[n // 2][n // 2])
